Tests/Shared/Mocks/EventBusHandler.swift

```diff
@@ -0,0 +1,62 @@
+import CioInternalCommon
+
+/// Mock of the EventBusHandler class, designed to mimic AutoMockable
+/// Once the class is generated using AutoMockable, it should seamlessly replace the current implementation without any issues
+public class EventBusHandlerMock: EventBusHandler, Mock {
```

Comment:
Due to generics, I couldn't generate it using `AutoMockable`, but I tried to keep the implementation similar to it. If we can generate this using `AutoMockable`, that should hopefully work without making any change in tests.

Reply:
i think it's a fair call, using generic here really helps with the casting of events when mapped from map

Reply:
Agreed